### ISSUE TITLE
EnTK hangs on Summit for MDFF

### ISSUE THREAD

**Issue by benjha** [radical-collaboration/MDFF-EnTK#9]
Hi @lee212 

How fast the simple mddf sample should run with enTK ?

I waited like 30 min. in the first task when decided to kill the job:

```
(radical_tools_python) [benjha@login5.summit mdff]$ python mdff.py 
EnTK session: re.session.login5.benjha.018293.0000
Creating AppManagerSetting up RabbitMQ system                                 ok
                                                                              ok
Validating and assigning resource manager                                     ok
Setting up RabbitMQ system                                                   n/a
new session: [re.session.login5.benjha.018293.0000]                            \
database   : [mongodb://rct:rct_test@two.radical-project.org/rct_test]        ok
create pilot manager                                                          ok
submit 1 pilot(s)
        [ornl.summit:168]
                                                                              ok
All components created
Update: simple-mdff state: SCHEDULING
Update: simple-mdff.Generating a simulated density map state: SCHEDULING
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: SCHEDULING
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: SCHEDULED
Update: simple-mdff.Generating a simulated density map state: SCHEDULED
create unit manager/gpfs/alpine/world-shared/bip115/radical_tools_python/lib/python3.7/site-packages/pymongo/topology.py:155: UserWarning: MongoClient opened before fork. Create MongoClient only after forking. See PyMongo's documentation for details: http://api.mongodb.org/python/current/faq.html#is-pymongo-fork-safe
  "MongoClient opened before fork. Create MongoClient only "
                                                           ok
add 1 pilot(s)                                                                ok
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok      <- enTK didn't pass this point
```

Could you take a look to the code:

```
/gpfs/alpine/world-shared/bip115/enTK_tests/mdff/mdff.py
```
and the installation:
```
/gpfs/alpine/world-shared/bip115/radical_tools_python
```
to make sure everything is correct ?

To activate the environment:
```
module load python/3.7.0
source  /gpfs/alpine/world-shared/bip115/radical_tools_python/bin/activate
```
There is also another test, based on an enTK example, that got till the end of the workflow:
```
/gpfs/alpine/world-shared/bip115/enTK_tests/eop/eop.py
```

Thanks!

Benjamin

**Comment by lee212** [comment]
Hi Ben,

Thank you for reporting this.

The numbers of cpu cores allocated to simulation tasks should be smaller than requested cpu cores, more specifically, the dict for workflow would work:
```
workflow_config = { 'ornl_summit' : 
                   { 'simulation': 
                    { 'cpus': 160 },
                     'analysis':
                     { 'cpus': 1 }}}
```
if you reduce 168 to 160 at the line 21 of your `mdff.py` script. Could you give it a try?

**Comment by lee212** [comment]
@benjha , do you need support on this trial? I know that we've exchanged email messages w.r.t. this test run on Summit. I myself tried the simple MDFF with your NAMD/VMD installation which seemed working fine. BTW, I am just starting to work on the Conda package for EnTK on PPC machines. I will create a separate ticket for Conda support.

**Comment by benjha** [comment]
Hi @lee212, 

Your suggestion partially fix the problem. The next hang occurred when trying to wget the pdb files. Compute nodes have no access to the external world.

So I needed to download the pdb files manually in the same directory  ```mdff.py``` lives and remove the wget tasks from the script. After that, the execution  worked, but some tasks failed (this output is similar than the one you posted from your jupyter notebook):


```
EnTK session: re.session.login5.benjha.018313.0003
Creating AppManagerSetting up RabbitMQ system                                 ok
                                                                              ok
Validating and assigning resource manager                                     ok
Setting up RabbitMQ system                                                   n/a
new session: [re.session.login5.benjha.018313.0003]                            \
database   : [mongodb://rct:rct_test@two.radical-project.org/rct_test]        ok
create pilot manager                                                          ok
submit 1 pilot(s)
        [ornl.summit:168]
                                                                              ok
All components created
Update: simple-mdff state: SCHEDULING
Update: simple-mdff.Generating a simulated density map state: SCHEDULING
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: SCHEDULING
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: SCHEDULED
Update: simple-mdff.Generating a simulated density map state: SCHEDULED
create unit manager/gpfs/alpine/world-shared/bip115/radical_tools_python/lib/python3.7/site-packages/pymongo/topology.py:155: UserWarning: MongoClient opened before fork. Create MongoClient only after forking. See PyMongo's documentation for details: http://api.mongodb.org/python/current/faq.html#is-pymongo-fork-safe
  "MongoClient opened before fork. Create MongoClient only "
                                                           ok
add 1 pilot(s)                                                                ok
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: EXECUTED
Update: simple-mdff.Generating a simulated density map.Starting to load the target PDB state: FAILED
Update: simple-mdff.Generating a simulated density map state: DONE
Update: simple-mdff.Converting the density map to an MDFF potential state: SCHEDULING
Update: simple-mdff.Converting the density map to an MDFF potential.generate dx file state: SCHEDULING
Update: simple-mdff.Converting the density map to an MDFF potential.generate dx file state: SCHEDULED
Update: simple-mdff.Converting the density map to an MDFF potential state: SCHEDULED
Update: simple-mdff.Converting the density map to an MDFF potential.generate dx file state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.Converting the density map to an MDFF potential.generate dx file state: EXECUTED
Update: simple-mdff.Converting the density map to an MDFF potential.generate dx file state: FAILED
Update: simple-mdff.Converting the density map to an MDFF potential state: DONE
Update: simple-mdff.Preparing the initial structure state: SCHEDULING
Update: simple-mdff.Preparing the initial structure.Starting to load the initial structure state: SCHEDULING
Update: simple-mdff.Preparing the initial structure.Starting to load the initial structure state: SCHEDULED
Update: simple-mdff.Preparing the initial structure state: SCHEDULED
Update: simple-mdff.Preparing the initial structure.Starting to load the initial structure state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.Preparing the initial structure.Starting to load the initial structure state: EXECUTED
Update: simple-mdff.Preparing the initial structure.Starting to load the initial structure state: FAILED
Update: simple-mdff.Preparing the initial structure state: DONE
Update: simple-mdff.Defining secondary structure restraints state: SCHEDULING
Update: simple-mdff.Defining secondary structure restraints.task.0003 state: SCHEDULING
Update: simple-mdff.Defining secondary structure restraints.task.0003 state: SCHEDULED
Update: simple-mdff.Defining secondary structure restraints state: SCHEDULED
Update: simple-mdff.Defining secondary structure restraints.task.0003 state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.Defining secondary structure restraints.task.0003 state: EXECUTED
Update: simple-mdff.Defining secondary structure restraints.task.0003 state: FAILED
Update: simple-mdff.Defining secondary structure restraints state: DONE
Update: simple-mdff.Running the MDFF simulation with NAMD state: SCHEDULING
Update: simple-mdff.Running the MDFF simulation with NAMD.task.0004 state: SCHEDULING
Update: simple-mdff.Running the MDFF simulation with NAMD.task.0004 state: SCHEDULED
Update: simple-mdff.Running the MDFF simulation with NAMD state: SCHEDULED
Update: simple-mdff.Running the MDFF simulation with NAMD.task.0004 state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.Running the MDFF simulation with NAMD.task.0004 state: EXECUTED
Update: simple-mdff.Running the MDFF simulation with NAMD.task.0004 state: FAILED
Update: simple-mdff.Running the MDFF simulation with NAMD state: DONE
Update: simple-mdff.NAMD simulation state: SCHEDULING
Update: simple-mdff.NAMD simulation.task.0005 state: SCHEDULING
Update: simple-mdff.NAMD simulation.task.0005 state: SCHEDULED
Update: simple-mdff.NAMD simulation state: SCHEDULED
Update: simple-mdff.NAMD simulation.task.0005 state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.NAMD simulation.task.0005 state: EXECUTED
Update: simple-mdff.NAMD simulation.task.0005 state: FAILED
Update: simple-mdff.NAMD simulation state: DONE
Update: simple-mdff.Calculating the root mean square deviation state: SCHEDULING
Update: simple-mdff.Calculating the root mean square deviation.task.0006 state: SCHEDULING
Update: simple-mdff.Calculating the root mean square deviation.task.0006 state: SCHEDULED
Update: simple-mdff.Calculating the root mean square deviation state: SCHEDULED
Update: simple-mdff.Calculating the root mean square deviation.task.0006 state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.Calculating the root mean square deviation.task.0006 state: EXECUTED
Update: simple-mdff.Calculating the root mean square deviation.task.0006 state: FAILED
Update: simple-mdff.Calculating the root mean square deviation state: DONE
Update: simple-mdff.Calculating the cross-correlation coefficient state: SCHEDULING
Update: simple-mdff.Calculating the cross-correlation coefficient.task.0007 state: SCHEDULING
Update: simple-mdff.Calculating the cross-correlation coefficient.task.0007 state: SCHEDULED
Update: simple-mdff.Calculating the cross-correlation coefficient state: SCHEDULED
Update: simple-mdff.Calculating the cross-correlation coefficient.task.0007 state: SUBMITTING
submit 1 unit(s)
        .                                                                     ok
Update: simple-mdff.Calculating the cross-correlation coefficient.task.0007 state: EXECUTED
Update: simple-mdff.Calculating the cross-correlation coefficient.task.0007 state: FAILED
Update: simple-mdff.Calculating the cross-correlation coefficient state: DONE
Update: simple-mdff state: DONE
close unit manager                                                            ok
wait for 1 pilot(s)
              0                                                               ok
closing session re.session.login5.benjha.018313.0003                           \
close pilot manager                                                            \
wait for 1 pilot(s)
              0                                                          timeout
                                                                              ok
session lifetime: 302.5s                                                      ok
All components terminated
```

**Comment by benjha** [comment]
I re-run again the script WITH wget and didn't hang this time, but still some tasks' state is FAILED

**Comment by lee212** [comment]
Thanks for the quick update! I will need to look at your script to see why it's failed, which I believe it is `/gpfs/alpine/world-shared/bip115/enTK_tests/mdff/mdff.py`. I have my version but it is a same simple mdff that is located at https://github.com/radical-collaboration/MDFF-Error/blob/master/simple_mdff.summit.py. Do you mind to try this one?

**Comment by lee212** [comment]
My run completed with one error that I've seen NAMD library loading error: `/gpfs/alpine/world-shared/bip115/NAMD_binaries/summit/NAMD_build.latest/Linux-POWER-MPI-smp-Summit/namd2: error while loading shared libraries: libfftw3f.so.3: cannot open shared object file: No such file or directory`. Do you know which module did I forget to load prior to namd on Summit?

**Comment by benjha** [comment]
it should be     ```fftw/3.3.8```

```
module avail fftw

----- /autofs/nccs-svm1_sw/summit/modulefiles/site/linux-rhel7-ppc64le/spectrum-mpi/10.3.0.1-20190611-aqjt3jo/xl/16.1.1-3 ------
   fftw/3.3.8
```

**Comment by lee212** [comment]
It indeed solves my error, I added this module load to the entk script.

Thanks!

**Comment by lee212** [comment]
@benjha , Do you think you can confirm this resolves and runs fine also?

**Comment by benjha** [comment]
Yes, that version worked well.

**Comment by lee212** [comment]
I'll close this now and feel free to re-open/add comments anytime.
